### PR TITLE
Vuex added, store is set up with example global vars

### PR DESCRIPTION
**Pull request by davidzynda**
## What this PR changes/adds

_Briefly describe WHAT your pr changes, which features it adds/modifies._

## Why it does that

_Briefly state why the change was necessary._

## Further notes

_List other areas of code that have changed but are not necessarily linked to the main feature. This could be method
signature changes, package declarations, bugs that were encountered and were fixed inline, etc._

## Linked Issue(s)

Closes # <-- _insert Issue number if one exists_

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: src/components/Home.vue [hunk 1]
```diff
@@ -1,14 +1,24 @@
...
   name: 'HomeComponent',
   components: {
+        Spinner,
     BatteryPassport
   },
+  data() {
+return {
+  loading: true
+}
+  },
```
Comment (felixsteinke) [inline src/components/Home.vue:21]: Your indentations are wrong. Please make sure to setup and use your editor configuration. 

For example i use IntelliJ and there is a "Reformat Code" option that manages these things automatically.

This is an issue in several lines.

File: src/components/Login.vue [hunk 2]
```diff
@@ -72,12 +72,7 @@ import {MOCK_AUTH_URL} from "@/services/service.const";
 
 export default {
   name: "LoginPage",
-  data() {
-    return {
-      email: "",
-      password: "",
-    };
-  },
+
```
Comment (felixsteinke) [inline src/components/Login.vue:75]: To keep the code clean, new lines should give more overview within the code. This new line is within an object definition and should be removed.

File: src/components/Login.vue [hunk 3]
```diff
@@ -110,6 +105,22 @@ LogoBG
...
+},
+password: {
+
+  set(newPassword) {
+this.$store.commit('setPassword', newPassword)
+  }
+},
+
+
+},
```
Comment (felixsteinke) [inline src/components/Login.vue:123]: Like described in previous comments, please watch on your code style (indentions and unnecessary new lines).

File: src/components/BatteryPassport.vue [hunk 4]
```diff
@@ -1,83 +1,80 @@
...
+      v-model="selectedProvider"
+      placeholder="Select Battery Provider"
+      @change="getBatteriesbyProvider()"
+      data-cy="provider-select"
+    >
+      <option value="" disabled selected>Select Battery Provider...</option>
+      <option
+        v-for="provider in listProviders"
+        :value="provider.name"
+        v-bind:key="provider.id"
```
Comment (felixsteinke) [inline src/components/BatteryPassport.vue:17]: A lot of changes in the html are only related to changed __indentations__. You should really discuss a uniform editorconfig: https://editorconfig.org/ 

Most of the other comments also are related to that.

File: src/components/BatteryPassport.vue [hunk 5]
```diff
@@ -120,6 +114,7 @@ Header
           let name = JSON.parse(user).name;
           console.log("CurrentUser: ",user, role);
 
+
```
Comment (felixsteinke) [inline src/components/BatteryPassport.vue:117]: This is not a problem. But is really unnecessary in my opinion. 

Your editor should show you the local changes before the commit, in my opinion it is useful to look on all changes again, identify unnecessary changes like that and just remove/revert the change.

File: src/components/Header.vue [hunk 6]
```diff
@@ -98,11 +96,7 @@ export default {
       this.$router.push({ name: "ScanPassport" });
     },
   },
-  mounted() {
-    let user = localStorage.getItem("user-info");
-    this.username = JSON.parse(user).email;
-    this.role = JSON.parse(user).role;
-  },
+
```
Comment (felixsteinke) [inline src/components/Header.vue:99]: same as in the other comments

File: src/store/index.js [hunk 7]
```diff
@@ -1,14 +1,19 @@
-import { createStore } from 'vuex'
+import { createStore } from "vuex";
```
Comment (felixsteinke) [inline src/store/index.js:1]: I did never use vuex, but the latest versions of VueJS use [Pinia](https://pinia.vuejs.org/) for state management. 
But as long as it does what you want it should be fine.

Just to let you know an alternative. ([Comparison of Pinia vs Vuex](https://blog.logrocket.com/pinia-vs-vuex/))

**Review by felixsteinke — CHANGES_REQUESTED** [review]

**Comment by felixsteinke** [comment]
I had a lot of comments, mostly related to your IDE editor configuration. But especially the indentations are a sign of a dirty code. Because of that the review has requests for changes.

**Comment by felixsteinke** [comment]
Just to let you know, you now merged the `dev branch` in here and this PR is for the `main`. I dont know if this is intended.

File: .github/workflows/publish-docker-images.yml [hunk 8]
```diff
@@ -7,7 +7,7 @@ name: "Publish docker images"
 
 on:
   push:
-    branches: [ "main" ]
+    branches: [ "main", "feature/CMP-16", "feature/CMP-150" ]
```
Comment (felixsteinke) [inline .github/workflows/publish-docker-images.yml:10]: are this permanent branches? i thought they would only address single issues and therefore make sure you really want to put them into the action

File: src/components/BatteryPassport.vue [hunk 9]
```diff
@@ -70,101 +39,82 @@
...
-  created(){
+  created() {
     this.loading = false;
 
   },
   components: {
-     Spinner,
-Header
+    Spinner,
+    Header
```
Comment (felixsteinke) [inline src/components/BatteryPassport.vue:64]: indentions are correct now, make sure to keep an eye on unnecessary new lines (whitespace)

**Review by felixsteinke — APPROVED** [review]
Looks good.

File: .github/workflows/publish-docker-images.yml (hunk 8 above)
Comment (saudkhan116) [reply .github/workflows/publish-docker-images.yml:10]: these branches are just created temporarily in order to test the configuration in INT environment. They will be deleted once tests are done.

**Review by saudkhan116 — APPROVED** [review]
Looks fine to me. 
Please make sure to write a brief description in PR about what is implemented and what need to be done further related to the ticket.
Just a suggestion for future PR.

**Pull request closed without merging**